spec/controllers/cities_controller_spec.rb

```diff
@@ -1,31 +1,38 @@
 require 'rails_helper'
 
 RSpec.describe CitiesController, type: :controller do
-  let(:city) { FactoryBot.create(:city_with_valid_name) }
-
   let(:valid_session) { {} }
+  let(:city_with_valid_name) do
+    { name: 'Конотоп' }
+  end
+
+  let(:city_with_invalid_name) do
+    { name: '' }
+  end
+
+  let(:user_admin) { FactoryBot.create(:user_admin) }
```

Comment:
You should test all actions with user without admin access too. And check that he do not have access